t/Test12-mppnscatter.sh

```diff
@@ -20,131 +20,42 @@
-  ncgen -o fv_core.res.tile4.nc $top_srcdir/t/Test12-input/fv_core.res.tile4.ncl
-  ncgen -o fv_core.res.tile5.nc $top_srcdir/t/Test12-input/fv_core.res.tile5.ncl
-  ncgen -o fv_core.res.tile6.nc $top_srcdir/t/Test12-input/fv_core.res.tile6.ncl
-
-  cd ..
-
-#TO DO: For loop through tiles?
-#Tile1 
-#Split the file: 
-  run command mppncscatter -i 2 -j 3 -x 2 -y 12 input/fv_core.res.tile1.nc
-  [ "$status" -eq 0 ]
-
-#Combine the file: 
-  run command mppnccombine -64 fv_core.res.tile1.nc fv_core.res.tile1.nc.???? 
-  [ "$status" -eq 0 ]
-
-#Compare the two:
-#  run command nccmp -md fv_core.res.tile1.nc input/fv_core.res.tile1.nc
-#  [ "$status" -eq 0 ]
-
-#Tile2 
-#Split the file: 
-  run command mppncscatter -i 2 -j 3 -x 2 -y 12 input/fv_core.res.tile2.nc
-  [ "$status" -eq 0 ]
-
-#Combine the file: 
-  run command mppnccombine -64 fv_core.res.tile2.nc fv_core.res.tile2.nc.???? 
-  [ "$status" -eq 0 ]
-
-#Compare the two:
-#  run command nccmp -md fv_core.res.tile2.nc input/fv_core.res.tile2.nc
-#  [ "$status" -eq 0 ]
-
-#Tile3 
-#Split the file: 
-  run command mppncscatter -i 2 -j 3 -x 2 -y 12 input/fv_core.res.tile3.nc
-  [ "$status" -eq 0 ]
-
-#Combine the file: 
-  run command mppnccombine -64 fv_core.res.tile3.nc fv_core.res.tile3.nc.???? 
-  [ "$status" -eq 0 ]
-
-#Compare the two:
-#  run command nccmp -md fv_core.res.tile3.nc input/fv_core.res.tile3.nc
-#  [ "$status" -eq 0 ]
-
-#Tile4 
-#Split the file: 
-  run command mppncscatter -i 2 -j 3 -x 2 -y 12 input/fv_core.res.tile4.nc
-  [ "$status" -eq 0 ]
-
-#Combine the file: 
-  run command mppnccombine -64 fv_core.res.tile4.nc fv_core.res.tile4.nc.???? 
-  [ "$status" -eq 0 ]
-
-#Compare the two:
-#  run command nccmp -md fv_core.res.tile4.nc input/fv_core.res.tile4.nc
-#  [ "$status" -eq 0 ]
-
-#Tile5 
-#Split the file: 
-  run command mppncscatter -i 2 -j 3 -x 2 -y 12 input/fv_core.res.tile5.nc
-  [ "$status" -eq 0 ]
-
-#Combine the file: 
-  run command mppnccombine -64 fv_core.res.tile5.nc fv_core.res.tile5.nc.???? 
-  [ "$status" -eq 0 ]
+  mkdir output
+  test_file=fv_core.res.tile1.nc
+  ncgen -o input/$test_file $top_srcdir/t/Test12-input/${test_file}l
+}
+  
+setup ()
+{
+  BASE_TEST_DIR=$( pwd )
+  mkdir $( basename $BATS_TEST_FILENAME .sh )
+  cd  $( basename $BATS_TEST_FILENAME .sh )
+}
 
-#Compare the two:
-#  run command nccmp -md fv_core.res.tile5.nc input/fv_core.res.tile5.nc
-#  [ "$status" -eq 0 ]
+teardown () {
+  cd $BASE_TEST_DIR
+  rm -rf  $( basename $BATS_TEST_FILENAME .sh )
+}
 
-#Tile6 
-#Split the file: 
-  run command mppncscatter -i 2 -j 3 -x 2 -y 12 input/fv_core.res.tile6.nc
+@test "Test mppncscatter" {
+  prepare_input_data
+  
+  # Scatter the file
+  run command mppncscatter -i 2 -j 3 -x 2 -y 12 input/$test_file
   [ "$status" -eq 0 ]
 
-#Combine the file: 
-  run command mppnccombine -64 fv_core.res.tile6.nc fv_core.res.tile6.nc.???? 
+  # Combine the file:
+  run command mppnccombine -64 output/$test_file ${test_file}.????
```

Comment:
Your updates didn't change this, but I wonder why the nccmp's were commented out previously. It would be good to verify that`input/$test_file` is identical to `output/$test_file`. If we wanted to do that, we could remove the `-64` from the `mppnccombine` so that the output file would be the same format as the input file, then add a nccmp check right after:

```
#  Verify that the recombined file is identical to the unscattered file
run command nccmp -d input/$test_file output/$test_file
[ "$status" eq 0 ]

Reply:
@ceblanton I think some of the extra commenting out of tests was to get the tests to pass.  I agree we should revamp all the tests to ensure we are testing all the correct things, including additional options, build options, etc.

Please put in an issue to remind us.

Reply:
Good idea- done. In particular, currently-commented out test script sections should be reviewed (deleted, uncommented, or adapted) along with more general assessment and improvement.

https://github.com/NOAA-GFDL/FRE-NCtools/issues/74